framework/aster-frame/src/task/processor.rs

```diff
@@ -1,185 +1,143 @@
 // SPDX-License-Identifier: MPL-2.0
 
 use alloc::sync::Arc;
-use core::sync::atomic::{AtomicUsize, Ordering::Relaxed};
-
-use lazy_static::lazy_static;
 
 use super::{
-    scheduler::{fetch_task, GLOBAL_SCHEDULER},
-    task::{context_switch, TaskContext},
-    Task, TaskStatus,
+    preempt::{
+        activate_preemption, deactivate_preemption, in_atomic, is_preemptible, panic_if_in_atomic,
+    },
+    scheduler::{add_task, locked_global_scheduler, pick_next_task},
+    task::{context_switch, NeedResched, Task, TaskContext},
 };
-use crate::{cpu_local, sync::Mutex, trap::disable_local};
+use crate::{arch::timer::register_scheduler_tick, cpu_local, sync::SpinLock, trap::disable_local};
 
+#[derive(Default)]
 pub struct Processor {
     current: Option<Arc<Task>>,
-    idle_task_cx: TaskContext,
+    idle_task_ctx: TaskContext,
 }
 
 impl Processor {
-    pub fn new() -> Self {
+    pub const fn new() -> Self {
         Self {
             current: None,
-            idle_task_cx: TaskContext::default(),
+            idle_task_ctx: TaskContext::empty(),
         }
     }
-    fn get_idle_task_cx_ptr(&mut self) -> *mut TaskContext {
-        &mut self.idle_task_cx as *mut _
-    }
-    pub fn take_current(&mut self) -> Option<Arc<Task>> {
-        self.current.take()
+    fn idle_task_ctx_ptr(&mut self) -> *mut TaskContext {
+        &mut self.idle_task_ctx as *mut _
     }
     pub fn current(&self) -> Option<Arc<Task>> {
         self.current.as_ref().map(Arc::clone)
     }
     pub fn set_current_task(&mut self, task: Arc<Task>) {
-        self.current = Some(task.clone());
+        self.current = Some(task);
     }
 }
 
-lazy_static! {
-    static ref PROCESSOR: Mutex<Processor> = Mutex::new(Processor::new());
+// FIXME: remove lock on the processor
+cpu_local! {
+    static PROCESSOR: SpinLock<Processor> = SpinLock::new(Processor::new());
```

Comment:
Could you please assist in replacing the usage of `SpinLock` with `RefCell` in this code snippet? After thoroughly reviewing the utilization of `PROCESSOR` throughout the entire codebase, we have concluded that there is no need for a lock in this scenario. However, I am uncertain about the code's compilability when incorporating `RefCell`.

Naturally, it is acknowledged that `RefCell` may not be the most optimal choice, considering the runtime cost it incurs, even if the code is successfully compiled. If `RefCell` can be utilized, we can introduce a new abstraction in subsequent pull requests.